lib/srv/regular/sshserver.go

```diff
@@ -761,15 +761,16 @@ func (s *Server) serveAgent(ctx *srv.ServerContext) error {
 	}
 
 	// start an agent on a unix socket
-	agentServer := &teleagent.AgentServer{Agent: ctx.GetAgent()}
+	agentServer := &teleagent.AgentServer{Agent: ctx.Parent.GetAgent()}
 	err = agentServer.ListenUnixSocket(socketPath, uid, gid, 0600)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	ctx.SetEnv(teleport.SSHAuthSock, socketPath)
-	ctx.SetEnv(teleport.SSHAgentPID, fmt.Sprintf("%v", pid))
-	ctx.AddCloser(agentServer)
-	ctx.AddCloser(dirCloser)
+	ctx.Parent.SetEnv(teleport.SSHAuthSock, socketPath)
+	ctx.Parent.SetEnv(teleport.SSHAgentPID, fmt.Sprintf("%v", pid))
+	ctx.Parent.AddCloser(agentServer)
+	ctx.Parent.AddCloser(dirCloser)
+	ctx.SyncParentEnv()
```

Comment:
Why `SyncParentEnv` here? You're already setting environment variables on the parent above.